### DISCUSSION TITLE
How to commit and upload only the modified data files in a "dvc-tracked" folder

### DISCUSSION THREAD

**Discussion by imflash217** [treeverse/dvc#5662] [Help] [answered]
Hi all,
I am tracking a folder `./data`
I added a new filed or edited previously tracked files inside `./data` folder like:
`./data/img_1.jpg`
Then, I do `dvc add data` and `dvc push`.
It uploads all the files inside the already tracked folder `data`. I want dvc to upload only the **newly added/modified** files.
How do I do it? 

My data folder is huge and uploading and downloading **all files** insided it for every `dvc push` and `dvc pull` takes up a long time and bandwidth.

What am I doing wrong here?

Thanks.

**Comment by pared** [comment, marked as answer]
Hello, @imflash217!

If the `dvc` pushes all the data, it means that the initial version of the data has never been pushed to this remote - dvc first checks which images are already in your `remote` and pushes only those that are new. Or at least that how it should be. Is it different in your case? Or you actually just want the new data to be on the remote?

**Reply by imflash217** [reply]
I check again by making few changes in my commits and saw exactly what you mentioned and how it should be expected. I was mistaken earlier because I was doing commits but forgetting to push...so was seeing the upload of all files at once.
My mistake.
Thanks @pared

**Reply by pared** [reply]
@imflash217 I am glad you found a solution!
